cw-orch-daemon/src/builder.rs

```diff
@@ -36,60 +31,89 @@ pub struct DaemonAsyncBuilder {
-        self.deployment_id = Some(deployment_id.into());
+    /// Specifies whether a fee grant should be used with this daemon
+    pub fn fee_granter(&mut self, granter: impl ToString) -> &mut Self {
+        self.sender_options.set_fee_granter(granter);
+        self
+    }
+
+    /// Specifies the hd_index of the daemon sender
+    pub fn hd_index(&mut self, index: u32) -> &mut Self {
+        self.sender_options.hd_index = Some(index);
         self
     }
 
     /// Set the mnemonic to use with this chain.
-    /// Defaults to env variable depending on the environment.  
+    /// Defaults to env variable depending on the environment.
     ///
     /// Variables: LOCAL_MNEMONIC, TEST_MNEMONIC and MAIN_MNEMONIC
     pub fn mnemonic(&mut self, mnemonic: impl ToString) -> &mut Self {
-        self.sender = Some(SenderBuilder::Mnemonic(mnemonic.to_string()));
-        self
-    }
-
-    /// Specifies a sender to use with this chain
-    /// This will be used in priority when set on the builder
-    pub fn sender(&mut self, wallet: Sender<All>) -> &mut Self {
-        self.sender = Some(SenderBuilder::Sender(wallet));
+        self.sender_options.mnemonic = Some(mnemonic.to_string());
         self
     }
+}
 
-    /// Specifies whether authz should be used with this daemon
-    pub fn authz_granter(&mut self, granter: impl ToString) -> &mut Self {
-        self.sender_options.set_authz_granter(granter);
+impl<SenderGen: SenderTrait> DaemonAsyncBuilderBase<SenderGen> {
+    /// Set the chain the daemon will connect to
+    pub fn chain(&mut self, chain: impl Into<ChainInfoOwned>) -> &mut Self {
+        self.chain = Some(chain.into());
         self
     }
 
-    /// Specifies whether a fee grant should be used with this daemon
-    pub fn fee_granter(&mut self, granter: impl ToString) -> &mut Self {
-        self.sender_options.set_fee_granter(granter);
+    /// Set the deployment id to use for the daemon interactions
+    /// Defaults to `default`
+    pub fn deployment_id(&mut self, deployment_id: impl Into<String>) -> &mut Self {
+        self.deployment_id = Some(deployment_id.into());
         self
     }
 
-    /// Specifies the hd_index of the daemon sender
-    pub fn hd_index(&mut self, index: u32) -> &mut Self {
-        self.sender_options.hd_index = Some(index);
-        self
+    /// Specifies a sender to use with this chain
+    /// This will be used in priority when set on the builder
+    pub fn sender<OtherSenderGen: SenderTrait>(
+        &self,
+        wallet: OtherSenderGen,
+    ) -> DaemonAsyncBuilderBase<OtherSenderGen> {
+        DaemonAsyncBuilderBase {
+            chain: self.chain.clone(),
+            deployment_id: self.deployment_id.clone(),
+            state_path: self.state_path.clone(),
+            state: self.state.clone(),
+            sender: Some(wallet),
+            sender_options: OtherSenderGen::SenderOptions::default(),
+            write_on_change: self.write_on_change,
+        }
```

Comment:
I can't mutate, because it's another object so I can't change its type. I can't take `self` either because it will throw a temporary object error. That's the only solution that we have with our `&mut self`-> `&mut Self` builder pattern